### PR TITLE
#163166265 Build a RestFul Endpoint To Update a Meetup 

### PR DESCRIPTION
**Pull request by Ogollah**
### What does this PR do?
Adds update meetup endpoint  for  QUESTIONER API

### Description of Task to be completed?
Have the following endpoints working
```
|Endpoint                            | Functionality                    |HTTP method 
|------------------------------------|----------------------------------|-------------
|/api/v1/user/signup                 |Signup account                    |POST        
|/api/v1/user/auth/signin            |Signin in user                    |POST
|/api/v1/user/auth/signout           |Signout in user                   |POST
|/api/v1/meetups/create              |Create a meetup record            |POST
|/api/v1/meetups/upcoming            |Get upcoming meetups              |GET
|/api/v1/meetups/<meetup-id>         |Fetch specific upcoming meetup    |GET
|/api/v1/questions/questions         |Fetch all question                |GET
|/api/v1/questions/<meetup-id>/create|Create a question                 |POST
|/api/v1/questions/<question-id> |access specific question                 |GET
|/api/v1/questions/<question-id>/upvote |upvote specific question        |PATCH
|/api/v1/questions/<question-id>/downvote |downvote specific question   |PATCH
|/api/v1/questions/<meetup-id>/rsvp       |Rsvp a specific meetup            |POST
|/api/v1/meetups/<meetup-id>/update       |Update specific upcoming meetup   |PATCH
```

### How should this be manually tested?
Using your terminal run the following commands:
```
- cd path/to/server-root
- git clone this repo
- cd to Questioner
- python3 -m venv env
- source env/bin/activate
- pip install requirement.txt
- on command line run manage.py test or run
```
### Any background context you want to provide?
No database
### What are the relevant pivotal tracker stories?
[#163166265](https://www.pivotaltracker.com/story/show/163166265)
[#163165499](https://www.pivotaltracker.com/story/show/163165499)

### Screen Shot?

updateddd

### CODE REVIEW

**Comment by Mupa1** [comment]
Looks good

**Comment by Ogollah** [comment]
Thanks @Mupa1

File: api/v1/main/controller/meetup_controller.py [hunk 1]
```diff
@@ -56,11 +62,52 @@ def get(self):
...
     @api.marshal_list_with(meetup)
     @jwt_required
     def get(self, meetup_id):
         """Get a specific meetup
         """
         return accessing_meetup(meetup_id)
+
         
+
+@api.errorhandler(NoAuthorizationError)
```
Comment (oagutu) [inline api/v1/main/controller/meetup_controller.py:79]: Why so many decorators? For `errorhandlers` all you need to do is initialise when the app is starting and this should take care of defined errors within your flask context

File: api/v1/main/service/meetup_service.py [hunk 2]
```diff
@@ -110,4 +110,73 @@ def get_all_meetups():
...
+    if meetup:
+        if admin:
+            meetup.topic=topic
+            meetup.description=description
+            meetup.images=images
+            meetup.Tags=Tags
+            meetup.createdOn=createdOn
+            happeningOn=happeningOn
+
+            response_object = {
```
Comment (oagutu) [inline api/v1/main/service/meetup_service.py:197]: Include the newly updated record data in your response

**Review by oagutu — COMMENTED** [review]
Good job @Ogollah . I have noted a few things:
 - Can’t update `happeningOn` or `host` values. Is this intentional?
- Able to update some fields with an empty string eg,  `topic, description`.  Please validate data to be inputted/
- You should be able to only provide fields to be updated when updating and not every single field for meeetup records

**Pull request merged**